File: iOS/Code/Swift/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtTitlePlaceholderTableViewCell.swift
//
//  BtTitlePlaceholderTableViewCell.swift
//  bugTrap
//
//  Created by Colby L Williams on 8/22/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

class BtTitlePlaceholderTableViewCell: BtNewBugDetailsTableViewCell {
	
	@IBOutlet weak var TitleLabel: UILabel!
	@IBOutlet weak var DetailLabel: UILabel!
	@IBOutlet weak var PlaceholderLabel: UILabel!
	
	override func setData(title: String? = nil, detail: String? = nil, placeholder: String? = nil, auxiliary: String? = nil) {
		TitleLabel.text = detail == nil || detail!.isEmpty ? "" : title
		DetailLabel.text = detail ?? ""
		PlaceholderLabel.text = detail == nil || detail!.isEmpty ? auxiliary ?? "" : ""
	}
}

File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Controllers/TableViewControllers/TableViewCells/BtTitlePlaceholderTableViewCell.cs
using System;

using UIKit;

namespace bugTrapKit
{
	public partial class BtTitlePlaceholderTableViewCell : BtNewBugDetailsTableViewCell
	{
		public BtTitlePlaceholderTableViewCell (IntPtr handle) : base(handle)
		{
			
		}

		public override void SetData (CellData cellData)
		{
			var title = cellData.Get(DataKeys.Title);
			var detail = cellData.Get(DataKeys.Detail);
			var auxiliary = cellData.Get(DataKeys.Auxiliary);

			TitleLabel.Text = string.IsNullOrEmpty(detail) ? detail : title;
			DetailTextLabel.Text = detail;
			PlaceholderLabel.Text = string.IsNullOrEmpty(detail) ? auxiliary : null;
		}
	}
}